lib/http.c

```diff
@@ -3770,8 +3771,8 @@ CURLcode Curl_http_readwrite_headers(struct Curl_easy *data,
       return result;
 
     /* Check for Content-Length: header lines to get size */
-    if(!k->ignorecl && !data->set.ignorecl &&
-       checkprefix("Content-Length:", k->p)) {
+    if(!k->httpcode_1xx && k->httpcode != 204 && k->httpcode != 304 &&
```

Comment:
I don't know if we can get away with this. Some servers flout the RFC and return body anyway.

Reply:
Even if they break the RFC in that way I think we could still get away with not reading the content-related headers. Especially if it makes us handle compliant responses better.